docs/delegate.md

```diff
@@ -38,16 +51,29 @@ should return:
 Note, specifying a port is optional. If no port is specified, then it defaults
 to 8448.
 
-With .well-known delegation, federating servers will check for a valid TLS
-certificate for the delegated hostname (in our example: `synapse.example.com`).
+### Serving a `.well-known/matrix/server` file with Synapse
```

Comment:
Up to you, but I might have been tempted to put this section first since it's the 'easy' case.

Reply:
I decided to put it second, because it's only useful in one very specific case. I thought it was clearer to put the more general solution first.

Reply:
That's reasonable! I think the way you introduced it will give enough hint that there are more options too, so no problem there